sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SpecificationBasedCheck.java

```diff
@@ -37,7 +40,8 @@ public abstract class SpecificationBasedCheck extends Check {
   private List<SecretMatcher> matcher;
 
   // shared map between all specificationBasedChecks used to calculate if secret was already reported for textRange
-  private Map<InputFileContext, List<TextRange>> reportedIssuesForCtx;
+  private Map<URI, List<TextRange>> reportedIssuesForCtx;
```

Comment:
SQ complained that `InputFileContext` is not `Comparable`, so I used the thing that plugin-api claims to be unique among the files

Reply:
Fine for me